lib/errors.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const { inherits, format } = require('util')
+const codes = {}
+
+/**
+ * ContentTypeParser
+*/
+createError('FST_ERR_CTP_ALREADY_PRESENT', `Content type parser '%s' already present.`)
+createError('FST_ERR_CTP_INVALID_TYPE', 'The content type should be a string', 500, TypeError)
+createError('FST_ERR_CTP_EMPTY_TYPE', 'The content type cannot be an empty string', 500, TypeError)
+createError('FST_ERR_CTP_INVALID_HANDLER', 'The content type handler should be a function', 500, TypeError)
+createError('FST_ERR_CTP_INVALID_PARSE_TYPE', `The body parser can only parse your data as 'string' or 'buffer', you asked '%s' which is not supported.`, 500, TypeError)
+createError('FST_ERR_CTP_BODY_TOO_LARGE', 'Request body is too large', 413, RangeError)
+createError('FST_ERR_CTP_INVALID_MEDIA_TYPE', `Unsupported Media Type: %s`, 415)
+createError('FST_ERR_CTP_INVALID_CONTENT_LENGTH', 'Request body size did not match Content-Length', 400, RangeError)
+
+function createError (code, message, statusCode = 500, Base = Error) {
+  if (!code) throw new Error('Fastify error code must not be empty')
+  if (!message) throw new Error('Fastify error message must not be empty')
+
+  code = code.toUpperCase()
+
+  function FastifyError (...args) {
```

Comment:
To achieve better performances we could avoid the spread operator and use directly the named parameters, I don't think they would change so much to justify the spread operator :)

Reply:
Do you have any numbers? I'm not convinced.

Reply:
I think @mcollina  and @davidmarkclements can give you a better answer here.

Reply:
Do we have a benchmark for this? Is this sitting in a hot path (like 404?)

Reply:
```

Reply:
Using spread operator has an impact of ~20% on my machine, woah! I tested the basic 404 case.

* with spread operator: 121K
* without arguments: 141K
* respect a fix count of args (3): 140K (current solution)